weedcoco/tests/repo/test_deposit.py

```diff
@@ -0,0 +1,124 @@
+import pathlib
+import json
+import pytest
+import os
+import filecmp
+import re
+from weedcoco.repo.deposit import main
+from weedcoco.validation import ValidationError
+
+TEST_DATA_DIR = pathlib.Path(__file__).parent / "deposit_data"
+TEST_DATA_SAMPLE_DIR = pathlib.Path(__file__).parent / "deposit_data_sample"
+TEST_BASIC_DIR_1 = TEST_DATA_DIR / "basic_1"
+TEST_BASIC_DIR_2 = TEST_DATA_DIR / "basic_2"
+TEST_COMPLETE_DIR = TEST_DATA_DIR / "complete"
+TEST_DUPLICATE_DIR = TEST_DATA_DIR / "duplicate"
+
+
+@pytest.fixture
+def executor(tmpdir):
+    test_repo_dir = tmpdir / "test_repo"
+
+    class Executor:
+        def run(self, weedcoco_path, image_dir, repository_dir=test_repo_dir):
+            args = [
+                "--weedcoco-path",
+                weedcoco_path,
+                "--image-dir",
+                image_dir,
+                "--repository-dir",
+                repository_dir,
+            ]
+            args = [str(arg) for arg in args]
+            main(args)
+            return test_repo_dir
+
+    return Executor()
+
+
+def _cmp_files(dir1, dir2):
```

Comment:
A problem with this design is that when it fails, all you'll get it "True" or "False". That's going to be a big pain to debug. Renaming these to `assert_files_equal` and `assert_weedcoco_equal`, and correspondingly making them consist of a series of `assert` statements, will mean that pytest can do its best to show you a rich analysis of the expression causing the failure: it can show you the parts of dicts that are not equal in `assert actual_dict == expected_dict`, or the values of variables in a comparison expression (and other local variables).